subsys/bluetooth/host/direction.c

```diff
@@ -131,3 +229,63 @@ static int hci_df_set_conn_cte_tx_param(struct bt_conn *conn, uint8_t cte_types,
 
 	return err;
 }
+
+/* @brief Function initializes Direction Finding in Host
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+int le_df_init(void)
+{
+	uint8_t max_switch_pattern_len;
+	uint8_t switch_sample_rates;
+	uint8_t max_cte_len;
+	uint8_t num_ant;
+	int err;
+
+	err = hci_df_read_ant_info(&switch_sample_rates, &num_ant,
+			     &max_switch_pattern_len, &max_cte_len);
+	if (err) {
+		return err;
+	}
+
+	df_ant_info.max_switch_pattern_len = max_switch_pattern_len;
+	df_ant_info.switch_sample_rates = switch_sample_rates;
+	df_ant_info.max_cte_len = max_cte_len;
+	df_ant_info.num_ant = num_ant;
+
+	BT_INFO("DF initialized.");
+	return 0;
+}
+
+int bt_le_df_set_adv_cte_tx_params(struct bt_le_ext_adv *adv,
+				const struct bt_le_df_adv_cte_tx_params *params)
+{
+	__ASSERT_NO_MSG(adv);
+	__ASSERT_NO_MSG(params);
+
+	int err;
+
+	if (!BT_FEAT_LE_CONNECTIONLESS_CTE_TX(bt_dev.le.features)) {
+		return -ENOTSUP;
+	}
+
+	/* Check if BT_ADV_PARAMS_SET is set, because it implies the set
+	 * has already been created.
+	 */
+	if (!atomic_test_bit(adv->flags, BT_ADV_PARAMS_SET)) {
+		return -EINVAL;
+	}
+
+	if (atomic_test_bit(adv->flags, BT_PER_ADV_CTE_ENABLED)) {
```

Comment:
I cannot see that this flag is set anywhere.

Reply:
The flag would be set in function that handles CTE enable HCI command. It will be delivered in following PR.

Reply:
@ppryga  The check of this flag should be moved to that commit as well then.

Reply:
Test of flag removed.